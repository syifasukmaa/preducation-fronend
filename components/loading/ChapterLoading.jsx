import React from 'react';

export default function ChapterLoading() {
  return (
    <>
      <tr className="my-10">
        <td className="px-4 w-[5%]">
          <p className="py-1 my-4 animate-pulse bg-neutral-03 rounded-sm"></p>
        </td>
        <td className="px-4 w-[20%]">
          <div className="flex flex-col h-full justify-between">
            <p className="py-1 animate-pulse bg-neutral-03 rounded-sm"></p>
          </div>
        </td>
        <td className="px-4 w-[15%]">
          <p className="py-1 animate-pulse bg-neutral-03 rounded-sm"></p>
        </td>
        <td className="px-4 w-[40%]">
          <div className="flex flex-col h-full justify-between">
            <p className="py-1 mt-1 w-5/6 animate-pulse bg-neutral-03 rounded-sm"></p>
            <p className="py-1 mt-1 w-3/4 animate-pulse bg-neutral-03 rounded-sm"></p>
            <p className="py-1 mt-1 w-4/5 animate-pulse bg-neutral-03 rounded-sm"></p>
          </div>
        </td>
        <td className="px-4 w-[40%]">
          <div className="flex flex-col lg:flex-row justify-start h-full">
            <p className="py-3 w-16 mt-2 xl:mt-0 mr-2 animate-pulse bg-neutral-03 rounded-xl"></p>
            <p className="py-3 w-16 mt-2 xl:mt-0 animate-pulse bg-neutral-03 rounded-xl"></p>
          </div>
        </td>
      </tr>
    </>
  );
}
