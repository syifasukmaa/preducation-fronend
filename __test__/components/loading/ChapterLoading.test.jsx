import { render } from '@testing-library/react'
import ChapterLoading from '@/components/loading/ChapterLoading'

describe('Chapter Loading', () => {
  it('renders loading skeleton', () => {
    const container = render(<ChapterLoading />)
    expect(container).toMatchSnapshot()
  })
})
