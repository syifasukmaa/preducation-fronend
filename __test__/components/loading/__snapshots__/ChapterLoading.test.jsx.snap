// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chapter Loading renders loading skeleton 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <tr
        class="my-10"
      >
        <td
          class="px-4 w-[5%]"
        >
          <p
            class="py-1 my-4 rounded-sm animate-pulse bg-neutral-03"
          />
        </td>
        <td
          class="px-4 w-[20%]"
        >
          <div
            class="flex flex-col justify-between h-full"
          >
            <p
              class="py-1 rounded-sm animate-pulse bg-neutral-03"
            />
          </div>
        </td>
        <td
          class="px-4 w-[15%]"
        >
          <p
            class="py-1 rounded-sm animate-pulse bg-neutral-03"
          />
        </td>
        <td
          class="px-4 w-[40%]"
        >
          <div
            class="flex flex-col justify-between h-full"
          >
            <p
              class="w-5/6 py-1 mt-1 rounded-sm animate-pulse bg-neutral-03"
            />
            <p
              class="w-3/4 py-1 mt-1 rounded-sm animate-pulse bg-neutral-03"
            />
            <p
              class="w-4/5 py-1 mt-1 rounded-sm animate-pulse bg-neutral-03"
            />
          </div>
        </td>
        <td
          class="px-4 w-[40%]"
        >
          <div
            class="flex flex-col justify-start h-full lg:flex-row"
          >
            <p
              class="w-16 py-3 mt-2 mr-2 xl:mt-0 animate-pulse bg-neutral-03 rounded-xl"
            />
            <p
              class="w-16 py-3 mt-2 xl:mt-0 animate-pulse bg-neutral-03 rounded-xl"
            />
          </div>
        </td>
      </tr>
    </div>
  </body>,
  "container": <div>
    <tr
      class="my-10"
    >
      <td
        class="px-4 w-[5%]"
      >
        <p
          class="py-1 my-4 rounded-sm animate-pulse bg-neutral-03"
        />
      </td>
      <td
        class="px-4 w-[20%]"
      >
        <div
          class="flex flex-col justify-between h-full"
        >
          <p
            class="py-1 rounded-sm animate-pulse bg-neutral-03"
          />
        </div>
      </td>
      <td
        class="px-4 w-[15%]"
      >
        <p
          class="py-1 rounded-sm animate-pulse bg-neutral-03"
        />
      </td>
      <td
        class="px-4 w-[40%]"
      >
        <div
          class="flex flex-col justify-between h-full"
        >
          <p
            class="w-5/6 py-1 mt-1 rounded-sm animate-pulse bg-neutral-03"
          />
          <p
            class="w-3/4 py-1 mt-1 rounded-sm animate-pulse bg-neutral-03"
          />
          <p
            class="w-4/5 py-1 mt-1 rounded-sm animate-pulse bg-neutral-03"
          />
        </div>
      </td>
      <td
        class="px-4 w-[40%]"
      >
        <div
          class="flex flex-col justify-start h-full lg:flex-row"
        >
          <p
            class="w-16 py-3 mt-2 mr-2 xl:mt-0 animate-pulse bg-neutral-03 rounded-xl"
          />
          <p
            class="w-16 py-3 mt-2 xl:mt-0 animate-pulse bg-neutral-03 rounded-xl"
          />
        </div>
      </td>
    </tr>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
